src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddOrRemoveVirtualKeyword.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+#nullable enable
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the 'virtual' keyword is not added to
+    /// or removed from a member.
+    /// </summary>
+    public class CannotAddOrRemoveVirtualKeyword : Rule
+    {
+        public override void Initialize(RuleRunnerContext context) => context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+
+        private void RunOnMemberSymbol(ISymbol left, ISymbol right, ITypeSymbol leftContainingType, ITypeSymbol rightContainingType, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            // Members must exist
+            if (left is null || right is null)
+            {
+                return;
+            }
```

Comment:
I don't think that this branch can ever be reached but I understand that you need to add this to suppress compiler nullable errors. Is there a way to declare that either left or right must be not null  via nullable parameter attributes?

Reply:
The second unit test (`EnsureNoCrashWhenMembersDoNotExist`) reaches this branch. If you have members with two different names, this branch gets hit.

Reply:
Oh I did misread that as an AND instead of an OR. I think there's a rule that will have such an issue but we can defer that discussion to when we enable nullable in the other rules.